tardis/gui.py

```diff
@@ -8,32 +41,61 @@
 from matplotlib.figure import *
 from matplotlib.backends.backend_qt4agg import FigureCanvasQTAgg as FigureCanvas
 from matplotlib.backends.backend_qt4 import NavigationToolbar2QT as NavigationToolbar
-import os
-if os.environ.get('QT_API', None) is None:
+if os.environ.get('QT_API', None)=='pyqt':
     from PyQt4 import QtGui, QtCore
-else:
+elif os.environ.get('QT_API', None)=='pyside':
     from PySide import QtGui, QtCore
-    
+else:
+    raise ImportError('QT_API was not set! Please exit the IPython console'+
+        ' and at the bash prompt use : \n\n export QT_API=pyside \n or \n'+
+        ' export QT_API=pyqt \n\n For more information refer to user guide.')
+import yaml
 from astropy import units as u
+
 from tardis import analysis, util
+from tardis import run_tardis
+import tardis
 
-# def current_ion_index(index, index_list):
-#     if not index in index_list:
-#         return None
-#     if not (index - 1) in index_list:
-#         return 0
-#     else:
-#         return current_ion_index(index - 1, index_list) + 1
-#
-# def current_ion_index(index, duplicate_list):
-#     if duplicate_list[index - 1] != duplicate_list[index]:
-#         return 0
-#     else:
-#         return current_ion_index(index - 1, duplicate_list) + 1
+if (parse_version(matplotlib.__version__)>=parse_version('1.4')):
+    matplotlib.style.use('fivethirtyeight')
+else:
+    print "Please upgrade matplotlib to a version >=1.4 for best results!"
+matplotlib.rcParams['font.family']='serif'
+matplotlib.rcParams['font.size']=10.0
+matplotlib.rcParams['lines.linewidth']=1.0
+matplotlib.rcParams['axes.formatter.use_mathtext']=True
+matplotlib.rcParams['axes.edgecolor']=matplotlib.rcParams['grid.color']
+matplotlib.rcParams['axes.linewidth']=matplotlib.rcParams['grid.linewidth']
+
+class TemporarilyUnavaliable(Exception):
+    """Exception raised when creation of active mode of tardis is attempted."""
+    
+    def __init__(self, value):
+        self.value = value
+    
+    def __str__(self):
+        return repr(self.value)
 
-class ModelViewer(QtGui.QWidget):
-    def __init__(self, parent=None):
-        # assumes that qt has already been initialized by starting IPython with the flag "--pylab=qt"
+class Tardis(QtGui.QMainWindow):
+    """Create the top level window for the GUI and wait for call to display 
+    data.
+    
+    Public methods:
+    --------------
+        1. __init__
+          Take the parent (default None), configuration yaml file (None) and 
+          the hdf atom data file (None) and creates all the widgets for the 
+          GUI. 
+
+        2.show_model 
+          Take the tardis model object and give it to the Tardis main window 
+          object for display, and call the show method to paint the main 
+          window.
```

Comment:
A new line here. Similarly to all other doc strings. Read this carefully https://www.python.org/dev/peps/pep-0257/